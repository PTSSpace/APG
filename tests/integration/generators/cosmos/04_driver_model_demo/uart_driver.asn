Module-uart-driver DEFINITIONS AUTOMATIC TAGS ::= BEGIN

  IMPORTS Cfe-tlm-header FROM Module-cfe-headers
          Real-range FROM Module-simple-types;



  Uart-driver-uart0-read-tlm ::= SEQUENCE { -- UART_DRIVER_UART_0_READ_MID uart_driver_ReadTlm_t (0x08D0)
    header Cfe-tlm-header (WITH COMPONENTS {
      primary (WITH COMPONENTS {
        packet-version-number (0),
        packet-identification (WITH COMPONENTS {
          packet-type-is-cmd (FALSE),
          sec-hdr-flag-is-present (TRUE),
          application-process-identifier (208)
        })
      })
    }),
    payload Payload-uart-driver
  }

  Uart-driver-uart1-read-tlm ::= SEQUENCE { -- UART_DRIVER_UART_1_READ_MID uart_driver_ReadTlm_t (0x08D1)
    header Cfe-tlm-header (WITH COMPONENTS {
      primary (WITH COMPONENTS {
        packet-version-number (0),
        packet-identification (WITH COMPONENTS {
          packet-type-is-cmd (FALSE),
          sec-hdr-flag-is-present (TRUE),
          application-process-identifier (209)
        })
      })
    }),
    payload Payload-uart-driver
  }



  Payload-uart-driver ::= SEQUENCE {
    raw Real-range -- [m] TODO String
  }

END
