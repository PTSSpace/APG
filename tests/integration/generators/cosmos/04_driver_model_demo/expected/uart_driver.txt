# This file was autogenerated from ASN.1 model.

TELEMETRY uart_driver Uart-driver-uart0-read-tlm BIG_ENDIAN "UART_DRIVER_UART_0_READ_MID uart_driver_ReadTlm_t (0x08D0)"
    APPEND_ID_ITEM      packet-version-number                 3 UINT          0 ""
    APPEND_ID_ITEM      packet-type-is-cmd                    1 UINT          0 ""
        STATE False                           0
        STATE True                            1
    APPEND_ID_ITEM      sec-hdr-flag-is-present               1 UINT          1 ""
        STATE False                           0
        STATE True                            1
    APPEND_ID_ITEM      application-process-identifier       11 UINT       0xD0 ""
    APPEND_ITEM         sequence-flags                        2 UINT            ""
        STATE continuation-packet-in-sequence 0
        STATE first-packet-in-sequence        1
        STATE last-packet-in-sequence         2
        STATE complete-packet                 3
    APPEND_ITEM         packet-sequence-count                14 UINT            ""
    APPEND_ITEM         packet-data-length                   16 UINT            "length of the packet"
    APPEND_ITEM         seconds                              32 UINT            ""
        UNITS sec sec
    APPEND_ITEM         subsecs                              16 UINT            "unit: 2**-16 sec = 0.015 ms"
    APPEND_ITEM         spare                                32 UINT            "Spares"
    APPEND_ITEM         raw                                  64 FLOAT           "TODO String"
        LIMITS DEFAULT 5 ENABLED -1e-40 -1e-40 1e+40 1e+40
        UNITS m m

TELEMETRY uart_driver Uart-driver-uart1-read-tlm BIG_ENDIAN "UART_DRIVER_UART_1_READ_MID uart_driver_ReadTlm_t (0x08D1)"
    APPEND_ID_ITEM      packet-version-number                 3 UINT          0 ""
    APPEND_ID_ITEM      packet-type-is-cmd                    1 UINT          0 ""
        STATE False                           0
        STATE True                            1
    APPEND_ID_ITEM      sec-hdr-flag-is-present               1 UINT          1 ""
        STATE False                           0
        STATE True                            1
    APPEND_ID_ITEM      application-process-identifier       11 UINT       0xD1 ""
    APPEND_ITEM         sequence-flags                        2 UINT            ""
        STATE continuation-packet-in-sequence 0
        STATE first-packet-in-sequence        1
        STATE last-packet-in-sequence         2
        STATE complete-packet                 3
    APPEND_ITEM         packet-sequence-count                14 UINT            ""
    APPEND_ITEM         packet-data-length                   16 UINT            "length of the packet"
    APPEND_ITEM         seconds                              32 UINT            ""
        UNITS sec sec
    APPEND_ITEM         subsecs                              16 UINT            "unit: 2**-16 sec = 0.015 ms"
    APPEND_ITEM         spare                                32 UINT            "Spares"
    APPEND_ITEM         raw                                  64 FLOAT           "TODO String"
        LIMITS DEFAULT 5 ENABLED -1e-40 -1e-40 1e+40 1e+40
        UNITS m m
